multiplication_models/functions_for_all_nutrients.py

```diff
@@ -0,0 +1,158 @@
+import pandas as pd, numpy as np
+from db_queries import get_ids, get_outputs, get_location_metadata, get_population, get_covariate_estimates
+from get_draws.api import get_draws
+import scipy.stats
+import scipy.integrate as integrate
+import matplotlib.pyplot as plt
+
+def generate_coverage_parameter_draws(df):
+    """This function is used to generate 1000 draws of nutrient/vehicle coverage parameters based on
+    the mean value and confidence intervals. This function assumes a normal distribution of uncertainty
+    within the confidence interval centered around the mean and is truncated at the bounds of 0 and 100%"""
+    data_frame = df.copy()
+    np.random.seed(11)
+    for i in list(range(0,1000)):
+        data_frame[f'draw_{i}'] = scipy.stats.truncnorm.rvs(data_frame.a,
+                                                            data_frame.b,
+                                                            data_frame.value_mean,
+                                                            data_frame.value_std) / 100
+    data_frame = (data_frame
+                  .set_index(['location_id'])
+                  .drop(columns=[c for c in data_frame.columns if 'draw' not in c
+                                and c not in ['location_id','value_description']]))
+    return data_frame
+
+
+def generate_overall_coverage_rates(filepath,
+                                    nutrient,
+                                    vehicle,
+                                    coverage_levels,
+                                    years,
+                                    location_ids,
+                                    subpopulations):
+    """This function generates baseline and counterfactual coverage rates of fortification for a specified
+    nutrient and vehicle pair. The baseline coverage rates are assumed to remain constant from over all of
+    the specified years. The alternative coverage rates are assumed to jump from the baseline rate in the
+    first specified year to either 20/50/80 percent (or the defined coverage levels) of the proportion
+    of the population consuming fortifiable/industrially produced vehicle in the second year of the specified
+    years and then remains constant for the remaining years.
+
+    The location_ids paramter can be set to 'all' or a list of specific location IDs.
+
+    The subpopulation parameter should be carefully evaluated and set to a list of the desired subpopulations
+    of interest that may be specific to women of reproductive age or children under 5, etc.
+    """
+
+    data = pd.read_csv(filepath)
+    if location_ids == 'all':
+        data = (data.loc[data.sub_population.isin(subpopulations)].drop_duplicates())
+    else:
+        data = (data.loc[data.location_id.isin(location_ids)]
+            .loc[data.sub_population.isin(subpopulations)].drop_duplicates())

```

Comment:
This logic would be clearer to me if you grouped it differently.  Perhaps @beatrixkh has already done this, so remind me if it doesn't change after her PR and we can talk through alternative ways to do it.